sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -118,6 +118,9 @@ class SessionCatalog(
   }
 
   def dropDatabase(db: String, ignoreIfNotExists: Boolean, cascade: Boolean): Unit = {
+    if (db == "default") {
```

Comment:
Added test cases to show the differences between Hive metastore and `InMemoryCatalog`

- InMemoryCatalog: https://github.com/gatorsmile/spark/blob/dropDefaultDB/sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala#L943

- Hive Metastore:
https://github.com/gatorsmile/spark/blob/dropDefaultDB/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala#L500